core/integration/directory_test.go

```diff
@@ -71,6 +72,48 @@ func TestDirectoryWithNewFile(t *testing.T) {
 	require.Equal(t, []string{"some-file"}, res.Directory.WithNewFile.Entries)
 }
 
+func TestDirectoryName(t *testing.T) {
```

Comment:
I'm wondering what happens if the `Directory` is created from `dag.Directory()`
Is that a case we would like to test?

What about a empty scratch dir?

Reply:
Added and split into #6439.